lib/ansible/plugins/inventory/foreman.py

```diff
@@ -0,0 +1,198 @@
+# Copyright (C) 2016 Guido Günther <agx@sigxcpu.org>, Daniel Lobato Garcia <dlobatog@redhat.com>
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: forman
+    plugin_type: inventory
+    short_description: foreman inventory source
+    description:
+        - Get inventory hosts from the foreman service.
+    options:
+      url:
+        description: TODO
+      user:
+        description: TODO
+      password:
+        description: TODO
+      ssl_verify:
+        description: TODO
+      group_prefix:
+        description: TODO
+'''
+import re
+from distutils.version import LooseVersion
+
+from ansible.errors import AnsibleError, AnsibleParserError
+from ansible.module_utils._text import to_bytes
+from ansible.plugins.inventory import BaseInventoryPlugin
+
+# 3rd party imports
+try:
+    import requests
+    if LooseVersion(requests.__version__) < LooseVersion('1.1.0'):
+        raise ImportError
+except ImportError:
+        raise AnsibleError('This script requires python-requests 1.1 as a minimum version')
+
+from requests.auth import HTTPBasicAuth
+
+
+class InventoryModule(BaseInventoryPlugin):
+    ''' Host inventory parser for ansible using foreman as source. '''
+
+    NAME = 'foreman'
+
+    def __init__(self):
+
+        super(InventoryModule, self).__init__()
+
+        # from config
+        self.foreman_url = None
```

Comment:
Is there a way to raise errors in Ansible when this attribute is None?

Reply:
I can `raise AnsibleOptionsError`